### DISCUSSION TITLE
URL handling in the SearchResult WebPart

### DISCUSSION THREAD

**Discussion by jochenfritz-22** [microsoft-search/pnp-modern-search#3870] [Q&A]
Hello, 
I have a question regarding the handling of URLs in the SearchResult WebPart. I’m addressing this question in this forum as my research has shown that it relates to how URLs are resolved in Modern Search.

Scenario: I have a list with entries, and each entry is assigned a link pointing to the intranet. The goal is to display all the list entries in a Search Result WebPart and make the link to the intranet accessible via the title.
For the corresponding column, I have created the “RefinableString123” which I can read in the Search Result WebPart.

Problem: In the Search Result WebPart, the string is altered in a way that it no longer remains a URL and, therefore, cannot be accessed. As seen in the example: URL in the PnP - Search Results / Debug

The URL could be modified using a regular expression, but since scripts are not allowed in the tenant, this is not an option.

I would greatly appreciate it if someone could provide a solution to this challenge.

Thank you in advance.
![PnP_MS](https://github.com/user-attachments/assets/08dc0a21-85fd-4e39-a01b-0352921ffa16)

**Comment by wobba** [comment]
Can you check using another tool that the value in the property has the spaces added and `:` removed? To ensure there is nothing weird in the web parts mangling this.

If the data somehow is stored this way, you can use handlebar helpers to manipulate the URL in your template. Remove spaces and replace // with ://.

**Reply by jochenfritz-22** [reply]
Hello wobba, SharePoint Online does not allow me the insertion and execution of custom JavaScript. I tried the following, but it doesn't work: 

`Handlebars.registerHelper('cleanURL', function(url) {
    let cleanedURL = url.replace(/\s+/g, '').replace(/^https?\/\/+/, 'http://');
    return cleanedURL;
});`

![Bildschirmfoto 2024-07-19 um 15 38 03](https://github.com/user-attachments/assets/a19ff807-8777-4401-ac38-b3a7c00188eb)

**Reply by wobba** [reply]
I didn't mean build one, I meant use one of the available ones https://microsoft-search.github.io/pnp-modern-search/extensibility/templating/#handlebars-helpers
